### ISSUE TITLE
Request: Turn on PSU when pressing "connect"

### ISSUE THREAD

**Issue by jbkuma** [kantlivelong/OctoPrint-PSUControl#96]
Along the lines "connect after turning on PSU" and "turn on when starting print" perhaps a feature that automatically turns on the PSU when the Connect button is pressed, with a settable delay to attempt the connection.

**Comment by kantlivelong** [comment]
I get the request in general but could you elaborate the exact use case here?

**Comment by jbkuma** [comment]
It's just a non-critical step saver.  Rather than having to hit the toggle then hit "connect to printer", one could click "connect to printer" and it would automatically toggle on the printer the attempt a connect after a brief delay. (or the power toggle could attempt the same)

**Comment by kantlivelong** [comment]
Yes but what is the use case here? Let me clarify, how do you have everything wired and set up? What steps does everything go through? Is your printer using USB serial? etc...

**Comment by jbkuma** [comment]
MP Select Mini which is USB Serial, I have the Pi a relay housed inside
cutting the DC to the printer.  Currently setup as NO, but I'm debating
switching it to it to NC so if/when I reboot the pi or restart octoprint it
won't cut power to the printer.  The power switch is wired ahead of the
relay and can cut power if necessary.

Jason "Kuma" Brinkerhoff
*Mad Science Workshoppe, *proprietor
http://jbkuma.com/workshoppe

**Comment by Gadroc** [comment]
This would be incredibly useful for me as well.  My setup is a TAZ6 which runs a RAMBo board.  I have a relay setup controlling power to the TAZ6.  Unfortunately auto connect via PortLister does not work in this scenario.  The chip running serial to USB on the Rambo board is powered via USB so the serial port is preset always while that USB cable is plugged in even though power is cut to the RAMBo board.

**Comment by blaskkaffe** [comment]
Would be very useful for me too!
I have the Pi connected to +5VSB and the controller card connected via USB, but it doesn't power from usb, so it is powered from 12v. It doesnt have the option to power both from usb and DC, only selectable via jumper (Anycubic Trigorilla).

So clicking the connect button to turn on psu and connect would definitely save a step!

**Comment by nionio6915** [comment]
blaskkaffe - have you looked at the Porlister plugin?

**Comment by saschaludwig** [comment]
+1 for the "connect after turning on PSU". I use octopi (always on) with an anycubic i3 mega on USB and it always require a manual "connect" click after switching on the PSU of the printer. Would be great to save that step, otherwise the "Automatically turn PSU ON" is rather useless because one always has to click connect to start the print

**Comment by nionio6915** [comment]
you need to look at portlister plugin. that is exactly what it does. it will connect to the printer when it comes online. you have to give a moment for portlister to find it. 
if you can not be patient for the watchdog cycle to complete, then you can connect it manually.

**Comment by Martyfree123** [comment]
Just download and use the Pull. That's what I've been doing and the auto connect printer option works perfect. If the creator of the plug in is unwilling to merger the Pull into the master then we'll just have to download and use the Pull.

**Comment by numanair** [comment]
Pull #101 seems to work perfectly for my use case, which sounds identical to @jbkuma's.
